File: 2.1. Two Pointers/4. Container With Most Water/dory.py
# Runtime 537ms
# Beats  26.54%

# Memory 30MB
# Beats 8.04%
class Solution:
    def maxArea(self, height: List[int]) -> int:
        result = 0
        left = 0
        right = len(height)-1

        while left < right:
            curr_size = (right - left) * min(height[left], height[right])
            result = max(curr_size, result)
            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        return result

File: 3.2. Sliding Window/1. Best Time to Buy And Sell Stock/dory.cs
# Runtime 1ms
# Beats 99.86%

# Memory 56.17MB
# Beats 83.84%

public class Solution {
    public int MaxProfit(int[] prices) {
        int result = 0;
        int min = int.MaxValue;
        foreach(int price in prices) {
            if(price < min) {
                min = price;
            }
            if(price - min >= result) {
                result = price - min;
            }
        }
        return result;
    }
}